contracts/src/v0.8/dev/automation/2_1/KeeperRegistryBase2_1.sol

```diff
@@ -794,7 +795,7 @@ abstract contract KeeperRegistryBase2_1 is ConfirmedOwner, ExecutionPrevention {
     UpkeepTransmitInfo memory transmitInfo
   ) internal returns (bool, bytes32) {
     LogTrigger memory trigger = abi.decode(rawTrigger, (LogTrigger));
-    bytes32 dedupID = keccak256(abi.encodePacked(upkeepId, trigger.txHash, trigger.logIndex));
+    bytes32 dedupID = keccak256(abi.encodePacked(upkeepId, trigger.logBlockHash, trigger.txHash, trigger.logIndex));
```

Comment:
This might introduce a new problem - it would allow the DON to submit multiple valid reports for the same log (txHash / logIndex) as long as it specifies different blockHashes. If the DON is careful and bug-free this shouldn't happen, but we are removing the on-chain protection against double upkeeps and pushing that responsibility to the DON. Is there a way we can preserve on-chain protections _and_ help the DON coordinate itself? Ex in the trigger we now include `logBlockHash` - can we look up this value and combine it with the dedupKey when trying to discern if a log has been handled on chain yet?

Reply:
Yep we can change this. This is currently where we read these logs https://github.com/smartcontractkit/chainlink/blob/develop/core/services/ocr2/plugins/ocr2keeper/evm21/upkeepstate/scanner.go#L64-L75 , right now we simply pass workID there, but if that doesn't match then we'll need to modify that logic

Reply:
discussed offline, this should be protected by reorg protection so okay to add to dedupe key